src/views/OtherStoresInventoryModal.vue

```diff
@@ -58,38 +122,201 @@ export default defineComponent({
+    this.sortByName();
   },
   methods: {
     closeModal() {
       modalController.dismiss({ dismissed: true });
     },
-    searchFacilities(){
+    searchFacilities() {
+      this.isRefreshing = true;
+      let filteredInventory = this.mapStock(this.storesInventory, this.storesInformation);
       if (this.queryString.trim() !== "") {
-        this.storesInventory = this.otherStoresInventory.filter((facility: any) => facility.facilityName.toLowerCase().includes(this.queryString.toLowerCase()));
+        filteredInventory = filteredInventory.filter((store: any) =>
+          store.storeName.toLowerCase().includes(this.queryString.toLowerCase())
+        );
+      }
+      if (this.hideEmptyStores) {
+        filteredInventory = filteredInventory.filter((store: any) => store.stock > 0);
+      }
+      this.storesWithInventory = filteredInventory;
+      this.isRefreshing = false;
+    },
+    // method to map the facility inventory from props to the store lookup data from api call
+    mapStock(storesInventory: any[], storesInformation: any[]): any[] {
+      return storesInformation.slice(1).map((store, index) => {  // skips first item (current facility)
+        const matchingFacility = storesInventory.find(facility => facility.facilityId === store.storeCode);
+        const storeHoursInfo = this.getStoreHoursStatus(store);
+        return {
+          ...store,
+          index: index + 1,
+          stock: matchingFacility ? matchingFacility.stock : 0,
+          hoursDisplay: storeHoursInfo.display,
+          isOpen: storeHoursInfo.isOpen
+        };
+      });
+    },
+    // logic for sorting stores by names
+    sortByNameLogic(a: any, b: any): number {
+      return a.storeName.localeCompare(b.storeName);
+    },
+    // method to sort the stores by distance 
+    sortByDistance() {
+      this.sortBy = 'distance';
+      // if store has Infinity distance, move it to the end
+      this.storesWithInventory.sort((a: any, b: any) => {
+        if (a.dist === "Infinity" && b.dist === "Infinity") return this.sortByNameLogic(a, b);  //If both stores have "Infinity" distance, sorts them alphabetically by store name
+        if (a.dist === "Infinity") return 1;  //If only store 'a' has "Infinity" distance, moves it to the end.
+        if (b.dist === "Infinity") return -1;  //If only store 'b' has "Infinity" distance, moves it to the end.
+        return a.dist - b.dist;  //For stores with numeric distances, sorts them in ascending order.
+      });
+    },
+    // method to sort the stores by name
+    sortByName() {
+      this.sortBy = 'name';
+      this.storesWithInventory.sort((a: any, b: any) => this.sortByNameLogic(a, b));
+    },
+    // method to run the current filter to maintain the current sort type when toggling empty stock toggle
+    runCurrentFilter() {
+      if (this.sortBy === 'distance') {
+        this.sortByDistance();
+      } else if (this.sortBy === 'name') {
+        this.sortByName();
+      }
```

Comment:
```suggestion
      } else {
        this.sortByName();
      }
```

No need for extra check as we only have two options